mcs/class/System/Test/System.Net/DnsTest.cs

```diff
@@ -239,21 +238,18 @@ public void GetHostByName ()
 			}
 		}
 
-		static IPAddress GetIPv4Address (IPHostEntry h)
+		static void CheckIPv4Address (IPHostEntry h, IPAddress a)
 		{
-			var al = h.AddressList.FirstOrDefault (x => x.AddressFamily == AddressFamily.InterNetwork);
-			if (al == null)
-				Assert.Ignore ("Could not resolve an IPv4 address as required by this test case, e.g. running on an IPv6 only network");
-			return al;
+			var al = h.AddressList.FirstOrDefault (x => x.AddressFamily == AddressFamily.InterNetwork && x.Equals (a));

```

Comment:
I can probably change this just to `.Contains`.

Reply:
Yeah, that'd be good.